AP_ADSB: Add Ping200x transponder interface via mission planner

nicholas-inocencio opened the pull request:
Codependent with https://github.com/ArduPilot/MissionPlanner/pull/2685. 

This allows the ardupilot to pass UCP/GDL90 transponder status messages from a connected ping200x down to the GCS via mavlink. 

The code should also be able to pass control messages via mavlink from the GCS up to the ping200x on UCP/GDL90, but there is an outstanding bug which causes those to fail; they are built properly, but _port->write() does not send the data across.

magicrub commented on libraries/AP_ADSB/AP_ADSB.cpp:
```diff
@@ -153,7 +153,7 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
     // @DisplayName: ADS-B logging
     // @Description: 0: no logging, 1: log only special ID, 2:log all
-    // @Values: 0:no logging,1:log only special ID,2:log all
+    // @Values: 0:disable GPS out,1:GCS Failsafe log only special ID,2:log all
```
Uhm.. you've updated the description to the wrong parameter :)

nicholas-inocencio replied:
Thanks for catching this; I have no idea how that happened.

magicrub commented on libraries/AP_ADSB/AP_ADSB.cpp:
```diff
@@ -632,6 +631,26 @@ void AP_ADSB::handle_out_cfg(const mavlink_uavionix_adsb_out_cfg_t &packet)
+void AP_ADSB::handle_out_control(const mavlink_uavionix_adsb_out_control_t &packet)
+{
+    out_state.ctrl.baroCrossChecked = packet.state & 1;
+    out_state.ctrl.airGroundState = packet.state & 4;
```
Please use (1L<<1) and  (1L<<2) ect ect . Muuuuch easier to read.

magicrub commented:
I'm on holiday right now but I'll make sure to take a closer look at this and the other ADS-B PRs next week.

magicrub commented:
@nicholas-inocencio this branch is based on an older PR that was changed a bit before it was merged. Can you please remove the merge and have a clean branch? It needs to get rebased due to conflicts

nicholas-inocencio commented:
> 
> 
> @nicholas-inocencio this branch is based on an older PR that was changed a bit before it was merged. Can you please remove the merge and have a clean branch? It needs to get rebased due to conflicts

Which PR/merge? Are you talking about the mavlink module?

nicholas-inocencio commented:
@magicrub Let me know if [this](https://github.com/nicholas-inocencio/ardupilot/commit/176e0e724fcdd88fbe5afd7248db50d8d9ecd01d) would be a better candidate for this PR. I think I've cleaned up the git history for it and kept the changes I require. Note that I removed some lines added by your [followup changes](https://github.com/ArduPilot/ardupilot/pull/18407/commits/f117355974fd1a35f412b9af681417e652a72337#).
